### PR TITLE
Improve template migrator capabilities

### PR DESCRIPTION
**Pull request by sergenyalcin**
### Description of your changes

Add Source, Target initializations and some common cli flags

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Tested against a few converters

### CODE REVIEW

File: converters/template/cmd/main.go [hunk 1]
```diff
@@ -49,4 +72,28 @@ func main() {
 
 	// Register all known API converters for the community AWS provider
 	provideraws.RegisterAllKnownConverters(registry)
+
+	// Initialize Source for reading resources
+	var source migration.Source
+	var err error
+	if len(*sourcePath) >= 1 { // FileSystem Source
```
Comment (erhancagirici) [inline converters/template/cmd/main.go:79]: - do we need a nil check here for `sourcePath`?
- nit comment, just asking for clarification: is it safe to assume omitting `sourcePath` directly means k8s source in terms of UX (e.g user wanted to use fs source but forgot to specify sourcePath and k8s source was used instead silently)?
- nit comment: why not `len(*sourcePath) > 0` :)

File: converters/template/cmd/main.go [hunk 2]
```diff
@@ -20,15 +20,38 @@
...
 func main() {
+	// Common CLI Flags
+	// They can be extended according to requirements
+	var (
+		app            = kingpin.New(filepath.Base(os.Args[0]), "Upbound migration plan generator for migrating Kubernetes objects from community providers to official providers.").DefaultEnvars()
+		planPath       = app.Flag("plan-path", "Path where the generated migration plan will be stored").Short('p').Default("migration_plan.yaml").String()
+		sourcePath     = app.Flag("source-path", "Path of the root directory for the filesystem source").Short('s').String()
+		kubeconfigPath = app.Flag("kubeconfig", "Path of the kubernetes config file. Defaults to ~/.kube/config ").String()
+	)
+	if len(*kubeconfigPath) == 0 {
```
Comment (erhancagirici) [inline converters/template/cmd/main.go:47]: same: do we need to do nil checks before dereferencing or do they have default values?

**Review by erhancagirici — COMMENTED** [review]
LGTM with a few notes

File: converters/template/cmd/main.go (hunk 1 above)
Comment (sergenyalcin) [reply converters/template/cmd/main.go]: - About nil check, the `kingpin` library converts nil values to empty string `""`. So without nil check we can use these values.
- For clarification, added a description to the `-s` flag.
- Done :)

File: converters/template/cmd/main.go (hunk 2 above)
Comment (sergenyalcin) [reply converters/template/cmd/main.go:47]: See above answer.

**Review by erhancagirici — APPROVED** [review]

**Pull request merged**